packages/flutter/lib/src/material/material_state.dart

```diff
@@ -104,10 +104,10 @@ typedef MaterialPropertyResolver<T> = T Function(Set<MaterialState> states);
 /// to provide a `defaultValue` to the super constructor, so that we can know
 /// at compile-time what its default color is.
 ///
-/// The [MaterialStateColor] class cannot universally be used in class
-/// constructors that were written to accept plain [Color] instances, and
-/// is intended to help bridge the gap between that original API and the newer
-/// MaterialStateProperty API.
+/// This class enables existing widget implementations with [Color]
+/// properties to be extended to also effectively support `MaterialStateProperty<Color>`
+/// property values. [MaterialStateColor] should only be used with widgets that document
+/// their support, like [ActionChip].
```

Comment:
I don't think ActionChip supports MaterialStateColor. TimePickerThemeData does: https://api.flutter.dev/flutter/material/TimePickerThemeData/hourMinuteTextColor.html

Reply:
Ohhh I see the difference.